src/Collapse/include/gudhi/Flag_complex_edge_collapser.h

```diff
@@ -15,366 +16,309 @@
+    init_neighbors_dense();
+#endif
+    std::vector<typename Ngb_list::sequence_type> neighbors2(num_vertices);
+    for(auto&&e : r){
+      using std::get;
+      Vertex u = get<0>(e);
+      Vertex v = get<1>(e);
+      Filtration_value f = get<2>(e);
+      neighbors2[u].emplace_back(v, f);
+      neighbors2[v].emplace_back(u, f);
+#ifdef GUDHI_COLLAPSE_USE_DENSE_ARRAY
+      neighbors_dense(u,v)=f;
+      neighbors_dense(v,u)=f;
+#endif
+    }
+    for(int i=0;i<neighbors2.size();++i){
+      neighbors2[i].emplace_back(i, -std::numeric_limits<Filtration_value>::infinity());
+      neighbors[i].adopt_sequence(std::move(neighbors2[i])); // calls sort
+#ifdef GUDHI_COLLAPSE_USE_DENSE_ARRAY
+      neighbors_dense(i,i)=-std::numeric_limits<Filtration_value>::infinity();
+#endif
     }
-    return edge_indices;
+    neighbors2.clear();
   }
 
-  // Detect and set all edges that are becoming critical
-  template<typename FilteredEdgeOutput>
-  void set_edge_critical(Edge_index indx, Filtration_value filt, FilteredEdgeOutput filtered_edge_output) {
-#ifdef DEBUG_TRACES
-    std::cout << "The curent index  with filtration value " << indx << ", " << filt << " is primary critical" <<
-    std::endl;
-#endif  // DEBUG_TRACES
-    std::set<Edge_index> effected_indices = three_clique_indices(indx);
-    // Cannot use boost::adaptors::reverse in such dynamic cases apparently
-    for (auto it = effected_indices.rbegin(); it != effected_indices.rend(); ++it) {
-      current_backward = *it;
-      Vertex_handle u = std::get<0>(f_edge_vector_[current_backward]);
-      Vertex_handle v = std::get<1>(f_edge_vector_[current_backward]);
-      // If current_backward is not critical so it should be processed, otherwise it stays in the graph
-      if (!critical_edge_indicator_[current_backward]) {
-        if (!edge_is_dominated(u, v)) {
-#ifdef DEBUG_TRACES
-          std::cout << "The curent index became critical " << current_backward  << std::endl;
-#endif  // DEBUG_TRACES
-          critical_edge_indicator_[current_backward] = true;
-          filtered_edge_output(u, v, filt);
-          std::set<Edge_index> inner_effected_indcs = three_clique_indices(current_backward);
-          for (auto inr_idx : inner_effected_indcs) {
-            if(inr_idx < current_backward) // && !critical_edge_indicator_[inr_idx]
-              effected_indices.emplace(inr_idx);
-          }
-#ifdef DEBUG_TRACES
-          std::cout << "The following edge is critical with filt value: {" << u << "," << v << "}; "
-            << filt << std::endl;
-#endif  // DEBUG_TRACES
-        }
+  // Open neighborhood
+  // At some point it helped gcc to add __attribute__((noinline)) here, otherwise we had +50% on the running time
+  // on one example. It looks ok now, or I forgot which example that was.
+  void common_neighbors(boost::container::flat_set<Vertex>& e_ngb, std::vector<std::pair<Filtration_value, Vertex>>& e_ngb_later, Vertex u, Vertex v, Filtration_value f_event){
```

Comment:
I can't find a good name. Something like Timed_vertices is less informative than spelling the type. Paraphrasing with Vector_of_filtration_and_vertex doesn't improve anything. Since it is only used twice, I think I'll leave it as it is.

Reply:
Ok let's keep it, but please let's keep the line less than 120 characters.